modeshape-common/src/main/java/org/modeshape/common/util/FileUtil.java

```diff
@@ -318,6 +318,20 @@ public static void addFileToZip(String path, String srcFile, ZipOutputStream zip
             }
         }
     }
+
+    /**
+     * Returns the extension of a file, including the dot.
+     * 
+     * @param filename the name of a file, may be not be null
+     * @return the file extension or an empty string if the extension cannot be determined.
+     */
+    public static String getExtension(final String filename) {
+        if (filename == null) {
+            throw new IllegalArgumentException("filename cannot be null");
+        }
```

Comment:
This is perfectly fine, but ICYMI also and for future reference, with Java 8 we can now do this using the built-in `java.util.Objects` class:

    Objects.requireNotNull(filename,"filename cannot be null");

Even better is that the methods return the parameter, so you can use it in constructors and method params quite easily to check and assign with a one-liner.